db/system_keyspace.cc

```diff
@@ -2642,6 +2654,310 @@ class clients_table : public streaming_virtual_table {
+    }
+
+    future<> tables(result_collector& result, const lw_shared_ptr<keyspace_metadata>& ks_meta) {
+        std::ostringstream os;
+        auto&& tables = ks_meta->tables();
+        boost::sort(tables, [] (const auto& a, const auto& b) {
+            return a->cf_name() < b->cf_name();
+        });
+
+        for (auto& t : tables) {
+            os.str("");
+            t->describe(_db, os, false);
+            auto desc = os.str();
+
+            os.str("");
+            t->describe(_db, os, true);
+            auto desc_internals = os.str();
+
+            co_await emit_row(result, "table", t->cf_name(), desc, desc_internals);
+        }
+    }
+
+    future<> types(result_collector& result, const lw_shared_ptr<keyspace_metadata>& ks_meta) {
+        auto& user_types_map = ks_meta->user_types().get_all_types();
+        auto user_types = boost::copy_range<std::vector<user_type>>(user_types_map | boost::adaptors::transformed([] (const auto& e) {
+            return e.second;
+        }));
+        boost::sort(user_types, [] (const auto& a, const auto& b) {
+            return a->get_name_as_cql_string() < b->get_name_as_cql_string();
+        });
+
+        for (auto& ut : user_types) {
+            auto desc = get_descriptions(*ut);
+            co_await emit_row(result, "type", ut->get_name_as_cql_string(), desc.first, desc.second);
+        }
+    }
+
+    future<> views(result_collector& result, const lw_shared_ptr<keyspace_metadata>& ks_meta) {
+        std::ostringstream os;
+        auto&& views = ks_meta->views();
+        boost::sort(views, [] (const auto& a, const auto& b) {
+            return a->cf_name() < b->cf_name();
+        });
+
+        for (auto& v : views) {
+            if (is_index(v->view_info()->base_id(), v)) {
+                continue;
+            }
+
+            os.str("");
+            v->describe(_db, os, false);
+            auto desc = os.str();
+
+            os.str("");
+            v->describe(_db, os, true);
+            auto desc_internals = os.str();
+
+            co_await emit_row(result, "view", v->cf_name(), desc, desc_internals);
+        }
+    }
+
+    future<> emit_keyspace_elements(result_collector& result, const sstring& ks) {
+        auto keyspace_meta = _db.find_keyspace(ks).metadata();
+
+        co_await aggregates(result, ks);
+        co_await functions(result, ks);
+        co_await indices(result, keyspace_meta);
+        co_await keyspace(result, keyspace_meta);
+        co_await tables(result, keyspace_meta);
+        co_await types(result, keyspace_meta);
+        co_await views(result, keyspace_meta);
+    }
+
+    future<> execute(reader_permit permit, result_collector& result, const query_restrictions& qr) override {
+        auto keyspaces = _db.get_all_keyspaces();
+        boost::sort(keyspaces);
+
+        for (auto& ks : keyspaces) {
+            auto dk = make_partition_key(ks);
```

Comment:
The sorting is done 3 lines above (`boost::sort(keyspaces);`) and below there is `if` that checks if the partition belongs to this shard and is present in the partition range. 
Should there be anything else?

Reply:
Oh, I've just now noticed that the partitions should be sorted according to their token value, not the cells' value. I'll fix this

Reply:
You can look at https://github.com/scylladb/scylladb/blob/fcf0628bc5d03f13d75f12e1e639747a178821bf/db/system_keyspace.cc#L2582-L2614
for an example of how this can be done.

Reply:
Now partition in both tables are emitted by their token values